database/pgsql/migrations/00001_initial_schema.go

```diff
@@ -14,179 +14,220 @@
 
 package migrations
 
-import "github.com/remind101/migrate"
+import "github.com/coreos/clair/pkg/dbutil"
 
-func init() {
-	RegisterMigration(migrate.Migration{
-		ID: 1,
-		Up: migrate.Queries([]string{
+var (
+	// entities are the basic building blocks to relate the vulnerabilities with
+	// the ancestry.
+	entities = dbutil.MigrationQuery{
```

Comment:
Entities are the *real* immutable content in the database.

On a higher level, Layer ( layer hash with layer content ) is not really immutable because a new extension can scan a layer and add new associated features.